dtasm3 fails on Rust dtasm modules

Currently, dtasm3 fails to run the Rust example modules, `make run-dtasm3` yields the following:
```
cd runtime/examples/dtasm3_main/build; ./dtasm3 ../../../../module/dpend_rs/target/wasm32-wasi/debug/dpend_rs.wasm
Model description verifies ok
ID: {8ad29d6a-2525-4576-95be-e1facde6860e}
Name: Double Pendulum
Description: Double pendulum simulation by solving equations of motion using a simple Runge-Kutta scheme (http://www.physics.usyd.edu.au/~wheat/dpend_html/solve_dpend.c)
Generating Tool: 
 can_handle_variable_step_size: 1
 can_interpolate_inputs: 0
 can_reset_step: 0
terminate called after throwing an instance of 'std::runtime_error'
  what():  Unknown dtasm status 100
Aborted (core dumped)
make: *** [Makefile:66: run-dtasm3] Error 134
```

~~Presumably, the reason might be that dtasm3 does not invoke `_initialize()` after loading these modules, which is a requirements for WASI reactor components.~~

--- a/runtime/dtasm3/src/dtasm3.cpp
+++ b/runtime/dtasm3/src/dtasm3.cpp
@@ -233,6 +233,8 @@ private:
             errMsg << "Response buffer too small; need " << resLen << " bytes, have " << m_buffersize;
             throw std::runtime_error(errMsg.str().c_str());
         }
+        // update memory base pointer to cover the case where wasm memory changed during function call
+        memPtr = m_m3Runtime.get_memory(memSize, 0);
 
         return memPtr + m_outputMem;
     }

